app/src/main/java/com/example/expenseappmvvm/screens/loginScreen/LoginViewModel.kt

```diff
@@ -28,7 +34,68 @@ class LoginViewModel(
     var email: MutableLiveData<String> = MutableLiveData()
     var password: MutableLiveData<String> = MutableLiveData()
 
-    fun onCreate() {}
+    var isValid = true
+
+    private lateinit var activity: LoginActivity
+
+    fun onCreate(parentActivity: LoginActivity) {
```

Comment:
you don't need the activity reference in view model, you can use a Single Live Data for navigation purpose